### ISSUE TITLE
[tool_crash] _TypeError: (#0      IsolateManager.initialize.<anonymous closure> (package:dds/src/isolate_manager.dart:241:32))

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#111926]
## Command
```
flutter run -d windows --profile
```

## Steps to Reproduce
1. ...
2. ...
3. ...

## Logs
_TypeError: (#0      IsolateManager.initialize.<anonymous closure> (package:dds/src/isolate_manager.dart:241:32))
```
#0      IsolateManager.initialize.<anonymous closure> (package:dds/src/isolate_manager.dart:241:32)
<asynchronous suspension>
#1      Mutex.runGuarded (package:dds/src/utils/mutex.dart:20:14)
<asynchronous suspension>
#2      IsolateManager.initialize (package:dds/src/isolate_manager.dart:217:5)
<asynchronous suspension>
#3      DartDevelopmentServiceImpl.startService (package:dds/src/dds_impl.dart:113:7)
<asynchronous suspension>
#4      DartDevelopmentService.startDartDevelopmentService (package:dds/dds.dart:97:5)
<asynchronous suspension>
#5

**Comment by jonahwilliams** [comment]
This happened while running a windows desktop app in profile mode, the app in question is:

```dart
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter/widgets.dart';

void main() =>
  runApp(
    Center(
      child:
        Image.network('https://pbs.twimg.com/profile_banners/284908325/1554440351/1500x500', loadingBuilder: (context, child, loadingProgress) {
          print('chunk!');
          return child;
        },)
      ),
    );
```

This may require the local changes I am testing, which are here: <TBD>

**Comment by jonahwilliams** [comment]
@bkonyi @christopherfujino

**Comment by jonahwilliams** [comment]
This doesn't reproduce each time, but as a hint, this code is spinning up a temporary isolate.

**Comment by jason-simmons** [comment]
This looks similar to https://github.com/flutter/flutter/issues/111482

The Dart DDS 2.3.0 package attempted to fix this, but it looks like it's still happening.

@bkonyi I think https://github.com/dart-lang/sdk/commit/db6311beca536bb58fc103c164b3320c1b3eb053 should be checking for a `type` instead of a `kind`

**Comment by bkonyi** [comment]
> @bkonyi I think [dart-lang/sdk@db6311b](https://github.com/dart-lang/sdk/commit/db6311beca536bb58fc103c164b3320c1b3eb053) should be checking for a `type` instead of a `kind`

Agh, you're right. Fix [up for review](https://dart-review.googlesource.com/c/sdk/+/260220).

**Comment by bkonyi** [comment]
Fix is published as `2.3.1`. This should be resolved (for real this time) once it rolls into the framework.

**Comment by christopherfujino** [comment]
Even though greg's commit accidentally closed this, it should indeed be closed, as 2.3.1 has rolled to the framework: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/pubspec.yaml#L13

**Comment by gspencergoog** [comment]
@christopherfujino I'm sorry, but how did I accidentally close this?  I haven't submitted anything to the SDK, and the referenced PR appears to be in my fork not the main repo, and there's only one auto-created PR in that fork that hasn't been submitted.

At about that time I did set up the build environment for the SDK on a machine and build it, but I just followed the setup instructions (and they don't include any pushing of PRs, AFAIK).
